workflow_tests/README.md

```diff
@@ -0,0 +1,422 @@
+const kieActJs = require('@kie/act-js');
+const kieMockGithub = require('@kie/mock-github');
+const utils = require('./utils');
+const assertions = require('./assertions/<workflow>Assertions');
+const mocks = require('./mocks/<workflow>Mocks');
+
+let mockGithub;
+const FILES_TO_COPY_INTO_TEST_REPO = [
+    {
+        src: path.resolve(__dirname, '..', '.github', 'actions'),
+        dest: '.github/actions',
+    },
+    {
+        src: path.resolve(__dirname, '..', '.github', 'libs'),
+        dest: '.github/libs',
+    },
+    {
+        src: path.resolve(__dirname, '..', '.github', 'scripts'),
+        dest: '.github/scripts',
+    },
+    {
+        src: path.resolve(__dirname, '..', '.github', 'workflows', '<workflow>.yml'),
+        dest: '.github/workflows/<workflow>.yml',
+    },
+];
+
+beforeEach(async () => {
+    mockGithub = new kieMockGithub.MockGithub({
+        repo: {
+            testWorkflowsRepo: {
+                files: FILES_TO_COPY_INTO_TEST_REPO,
+            },
+        },
+    });
+
+    await mockGithub.setup();
+});
+
+afterEach(async () => {
+    await mockGithub.teardown();
+});
+
+describe('test some general behaviour', () => {
+    test('something happens - test if expected happened next', async () => {
+        // get path to the local test repo
+        const repoPath = mockGithub.repo.getPath('testWorkflowsRepo') || '';
+
+        // get path to the workflow file under test
+        const workflowPath = path.join(repoPath, '.github', 'workflows', '<workflow>.yml');
+
+        // instantiate Act in the context of the test repo and given workflow file
+        let act = new kieActJs.Act(repoPath, workflowPath);
+
+        // set run parameters
+        act = utils.setUpActParams(
+            act,
+            '<event>',
+            {head: {ref: '<branch_name>'}},
+            {'<SECRET_NAME>': '<secret_value'},
+            '<github_token>',
+        );
+
+        // set up mocks
+        const testMockSteps = {
+            '<job_1_name>': [
+                {
+                    name: '<step_1_1_name>',
+                    mockWith: '<mock_command>',
+                },
+                {
+                    name: '<step_1_2_name>',
+                    mockWith: '<mock_command>',
+                },
+            ],
+            '<job_2_name>': [
+                utils.getMockStep('<step_2_1_name>', '<message>'),
+                utils.getMockStep('<step_2_2_name>', '<message>'),
+            ],
+        };
+
+        // run an event and get the result
+        const result = await act
+            .runEvent('<event>', {
+                workflowFile: path.join(repoPath, '.github', 'workflows'),
+                mockSteps: testMockSteps,
+            });
+
+        // assert results (some steps can run in parallel to each other so the order is not assured
+        // therefore we can check which steps have been executed, but not the set job order
+        assertions.assertSomethingHappend(result);
```

Comment:
```suggestion
        assertions.assertSomethingHappened(result);
```

Reply:
Updated in https://github.com/Expensify/App/pull/21937